Tests/KeyboardNavigatorTests.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  KeyboardNavigatorTests.swift
+//  KeyboardSupport-iOSTests
+//
+//  Created by Earl Gaspard on 8/3/18.
+//  Copyright © 2018 Bottle Rocket. All rights reserved.
+//
+
+import XCTest
+@testable import KeyboardSupport
+
+class KeyboardNavigatorTests: XCTestCase {
+
+    // MARK: - Properties
+    
+    private let keyboardToolbar = KeyboardToolbar()
+    
+    // MARK: - Tests
+    
+    func test_KeyboardNavigator_InitializesWithTextFields() {
+        let keyboardNavigator = keyboardNavigatorWithTextFields()
+        
+        XCTAssertEqual(keyboardNavigator.textInputs.count, 3)
+        XCTAssertEqual(keyboardNavigator.keyboardToolbar, keyboardToolbar)
+        XCTAssertTrue(keyboardNavigator.returnKeyNavigationEnabled)
+        XCTAssertEqual(keyboardNavigator.currentTextInputIndex, 0)
+    }
+    
+    func test_KeyboardNavigator_InitializesWithTextViews() {
+        let keyboardNavigator = keyboardNavigatorWithTextViews()
+        
+        XCTAssertEqual(keyboardNavigator.textInputs.count, 3)
+        XCTAssertEqual(keyboardNavigator.keyboardToolbar, keyboardToolbar)
+        XCTAssertTrue(keyboardNavigator.returnKeyNavigationEnabled)
+        XCTAssertEqual(keyboardNavigator.currentTextInputIndex, 0)
+    }
+    
+    func test_KeyboardNavigator_AccessoryDidTapBack() {
+        let textFieldNavigator = keyboardNavigatorWithTextFields()
+        textFieldNavigator.keyboardAccessoryDidTapNext(UIView())
+        textFieldNavigator.keyboardAccessoryDidTapBack(UIView())
+        
+        let textViewNavigator = keyboardNavigatorWithTextViews()
+        textViewNavigator.keyboardAccessoryDidTapNext(UIView())
+        textViewNavigator.keyboardAccessoryDidTapBack(UIView())
+    }
+    
+    func test_KeyboardNavigator_AccessoryDidTapNext() {
+        let textFieldNavigator = keyboardNavigatorWithTextFields()
+        textFieldNavigator.keyboardAccessoryDidTapNext(UIView())
+        
+        let textViewNavigator = keyboardNavigatorWithTextViews()
+        textViewNavigator.keyboardAccessoryDidTapNext(UIView())
+    }
+    
+    func test_KeyboardNavigator_AccessoryDidTapDone() {
+        let textFieldNavigator = keyboardNavigatorWithTextFields()
+        textFieldNavigator.keyboardAccessoryDidTapDone(UIView())
+        
+        let textViewNavigator = keyboardNavigatorWithTextViews()
+        textViewNavigator.keyboardAccessoryDidTapDone(UIView())
+    }
+}
+
+extension KeyboardNavigatorTests: KeyboardNavigatorDelegate {
```

Comment:
Is this up-to-date with the latest `2.0.2` version of KeyboardSupport? I thought @earlgaspard modified these to use a mock to validate the correct delegate method was being invoked. Actually, it looks like this file [isn't even in master](https://github.com/BottleRocketStudios/iOS-KeyboardSupport/tree/master/Tests). 🤔

Reply:
If I remember correctly this file was part of Earl's original 2.0 changes before he re-branched from master. I'll go through the two files and make sure we're not double-testing something.

Reply:
https://github.com/BottleRocketStudios/iOS-KeyboardSupport/pull/36/commits/75a67de95b85eb127a95037a4ccd7f6064b1b739

Reply:
These tests exercise the navigator delegate instead of the toolbar delegate.
I made a MockNavDelegate in the same style as the accessory one and switched to using that instead.